src/ledger/meta.zig

```diff
@@ -60,7 +60,7 @@ pub const SlotMeta = struct {
         };
     }
 
-    pub fn deinit(self: *Self) void {
+    pub fn deinit(self: Self) void {
```

Comment:
What is the reasoning behind passing by value rather than by reference? Honestly just curious here as my (probably naive) understanding tells me that for larger structs, passing by reference is usually preferred

Reply:
The main thing I was trying to do here was avoid needing a *mutable* pointer. That was an annoying constraint the caused issues in calling code that's unnecessary.

I passed by value instead of const pointer because the core zig developers recommend not to pass const pointers in cases like this, and instead you should pass by value allow the compiler to optimize this. For example: https://github.com/ziglang/zig/pull/19770

After our the recent discussions with the sig team though, I'm not really convinced by their arguments any more, and I think both approaches have their merits.